File: backend/Deliberation-session-ordinaire-service/src/main/java/com/example/deliberationsessionordinaireservice/models/Element.java
package com.example.deliberationsessionordinaireservice.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Element {
    Long id;

    String partieCours, partieTPs;

    Integer coefficientCours, coefficientTPs;

    Module module;
}

File: frontend/systeme-interne/src/app/models/PlanDeValidationModel/element.ts
import { Module } from "./module";

export class Element {
    id!:number;

    partieCours!: string;
    partieTPs!: string;

    coefficientCours!:number;
    coefficientTPs!:number;

    noteElement!:number;

    module!:Module;
}
